aten/src/ATen/native/Gesv.h

```diff
@@ -1,30 +1,92 @@
+#include <utility>
 #include "ATen/ATen.h"
 
 namespace at { namespace native {
 
-static inline void checkInputs(const Tensor& self, const Tensor& A) {
-  if (A.size(-1) != A.size(-2)) {
-    AT_ERROR("A must be batches of square matrices, "
-        "but they are %lld by %lld matrices",
-        (long long)A.size(-1), (long long)A.size(-2));
+static inline bool isTransposeContiguous(Tensor& self) {
+  return self.dim() == 2 &&
+          self.stride(0) == 1 &&
+          self.stride(1) == self.size(0);
+}
+
+/* gesv takes (self, A) and returns (sol, lu).
+ * (i)  output tensors (sol, lu) may be same as input tensors (self, A)
+ * (ii) for 2D matrices, .t_() represents their column-major format
+ *
+ * Cases:
+ * 1) `out` has correct shape but elements do not form a contiguous
+ * chunk of memory. Since shape is correct, we don't resize_ it. Instead, we
+ * clone the input tensor into a buffer, use the buffer for Lapack and finally
+ * copy the buffer to the output tensor.
+ *
+ * 2) out.t() is contiguous:
+ *    (i)  &in == &out: use out.data() as is. Do nothing
+ *    (ii) &in != &out: copy in.t() to out.t()
+ * 3) out.t() is not contiguous:
+ *    - resize_ should fix contiguity/size issues
+ *    (i)  &in == &out: copy in.t().clone() to out (same tensor)
+ *    (ii) &in != &out: copy in.t() to out
+ */
+static inline void prepareIOTensors(
```

Comment:
What does "prepareIOTensors" do? This comment gives a few cases but I'm not sure what this function is trying to accomplish (what is IO? input-output?) It would be great if this function name were more descriptive.

Reply:
how about `prepareTensorsForLapack`?

Reply:
That's better. What does it mean to prepare a tensor for lapack?

Reply:
basically, ensuring that pointers passed to Lapack represent Fortran contiguous tensors in column major format